test/SILOptimizer/OSLogFullOptTest.swift

```diff
@@ -269,5 +279,102 @@ func testDynamicPrecisionAndAlignment() {
+    // CHECK-NEXT: store i8 8, i8* [[OFFSET3]], align 1
+    
+    // CHECK: [[STR_POINTER:%.*]] = call swiftcc i8* @"${{.*}}getNullTerminatedUTF8Pointer{{.*}}"(i{{.*}} %0, {{.*}} %1
+
+    // os_log_impl call.
+    // CHECK-64: tail call swiftcc void @"${{.*}}_os_log_impl_test{{.*}}"({{.*}}, {{.*}}, {{.*}}, {{.*}}, i8* getelementptr inbounds ([17 x i8], [17 x i8]* @{{.*}}, i64 0, i64 0), i8* {{(nonnull )?}}[[BUFFER]], i32 12)
+    // CHECK-32: tail call swiftcc void @"${{.*}}_os_log_impl_test{{.*}}"({{.*}}, {{.*}}, {{.*}}, {{.*}}, i8* getelementptr inbounds ([17 x i8], [17 x i8]* @{{.*}}, i32 0, i32 0), i8* {{(nonnull )?}}[[BUFFER]], i32 8)
+    // CHECK-NEXT: [[BITCASTED_STR_STORAGE:%.*]] = bitcast i8* [[STR_STORAGE]] to %swift.opaque*
+    // CHECK-NEXT: tail call void @swift_arrayDestroy(%swift.opaque* [[BITCASTED_STR_STORAGE]]
+    // CHECK-NEXT: tail call void @swift_slowDealloc(i8* {{(nonnull )?}}[[STR_STORAGE]]
+    // CHECK-NEXT: tail call void @swift_slowDealloc(i8* {{(nonnull )?}}[[BUFFER]]
+    // CHECK: call void @llvm.lifetime.end{{.*}}({{.*}}, i8* {{(nonnull )?}}[[STR_STORAGE_PTR]]
+    // CHECK-NEXT: br label %[[EXIT]]
+  
+    // CHECK: [[EXIT]]:
+    // CHECK-NEXT: ret void
+}
+
+// CHECK-LABEL: define hidden swiftcc void @"${{.*}}testMetatypeInterpolation
+func testMetatypeInterpolation<T>(of type: T.Type) {
+    _osLogTestHelper("Metatype value: \(type)")
+    // CHECK: entry:
+    // CHECK: tail call swiftcc i1 @"${{.*}}isLoggingEnabled{{.*}}"()
+    // CHECK-NEXT: br i1 {{%.*}}, label %[[ENABLED:[0-9]+]], label %[[NOT_ENABLED:[0-9]+]]
+
+    // CHECK: [[NOT_ENABLED]]:
+    // CHECK-NEXT: call void @swift_release
+    // CHECK-NEXT: br label %[[EXIT:[0-9]+]]
+
+    // CHECK: [[ENABLED]]:
+    //
+    // Header bytes.
+    //
+    // CHECK-64: [[BUFFER:%.+]] = tail call noalias i8* @swift_slowAlloc(i{{.*}} 12
+    // CHECK-32: [[BUFFER:%.+]] = tail call noalias i8* @swift_slowAlloc(i{{.*}} 8
+    // CHECK-64-NEXT: [[STR_STORAGE:%.+]] = tail call noalias i8* @swift_slowAlloc(i{{.*}} 32
+    // CHECK-32-NEXT: [[STR_STORAGE:%.+]] = tail call noalias i8* @swift_slowAlloc(i{{.*}} 16
+    // CHECK: call void @llvm.lifetime.start{{.*}}({{.*}}, i8* {{(nonnull )?}}[[STR_STORAGE_PTR:%.*]]
+    // CHECK: store i8 2, i8* [[BUFFER]], align 1
+    // CHECK-NEXT: [[OFFSET1:%.+]] = getelementptr inbounds i8, i8* [[BUFFER]], i{{.*}} 1
+    // CHECK-NEXT: store i8 1, i8* [[OFFSET1]], align 1
+
+    // Argument bytes.
+    //
+    // CHECK-NEXT: [[OFFSET2:%.+]] = getelementptr inbounds i8, i8* [[BUFFER]], i{{.*}} 2
+    // TODO: Figure out why "store i8 16" turned into "store i8 0"
+    // CHECK-NEXT: store i8 32, i8* [[OFFSET2]], align 1
+    // CHECK-NEXT: [[OFFSET3:%.+]] = getelementptr inbounds i8, i8* [[BUFFER]], i{{.*}} 3
+    // CHECK-NEXT: store i8 8, i8* [[OFFSET3]], align 1
+    // CHECK-NEXT: [[TYPENAME:%.+]] = tail call swiftcc { i{{.*}}, %swift.bridge* } @"${{.*}}_typeName{{.*}}"({{.*}} %0
```

Comment:
It is great that we see a `_typeName(%0)` here. It would be good to add a case for constant meta types like `Int.self` and a FIXME there to indicate that Erik's optimization should apply.